src/metabase/analytics/prometheus.clj

```diff
@@ -258,6 +260,12 @@
   [metric]
   (some-> system .-registry metric prometheus/inc))
 
+(defn inc-and-throw!
```

Comment:
Ambivalent about this tiny wrapper. On the one hand, if we're going to use prometheus metrics more going forward, it will presumably be common to want to increment an error counter and throw an exception together. At the same time, it's not hard to wrap these two calls in a `do` block, and small wrappers like this can complicate linters or other static analysis tools that want to check, e.g., every place an exception is thrown.

Reply:
I like this. We might want to teach the linter that this function always throws.

Reply:
reading a bit more, i suspect we don't want an API like this. We have thoughtful error handling in certain places and we probably don't want to tie our hands with an overly simplistic api like this?

Reply:
I was also worried about `inc-and-throw!` cluttering up the stacktrace, so this is the first thing I checked, but it seems that Clojure attaches the stack at the point the `Throwable` is constructed, not at the point it is thrown, so the top frame will still be the callers frame.

```Clojure
dev> (defn call-inc-and-throw! []
       (prometheus/inc-and-throw! :foo (Exception. "foo")))
#'dev/call-inc-and-throw!
dev> (call-inc-and-throw!)
Execution error at dev/call-inc-and-throw! (REPL:1997).
foo
```

```
1. Unhandled java.lang.Exception
   foo

                      REPL: 1997  dev/call-inc-and-throw!
                      REPL: 1996  dev/call-inc-and-throw!
                      REPL: 1999  dev/eval209000
                      REPL: 1999  dev/eval209000
             Compiler.java: 7700  clojure.lang.Compiler/eval
    interruptible_eval.clj:  106  nrepl.middleware.interruptible-eval/evaluator/run/fn
    interruptible_eval.clj:  101  nrepl.middleware.interruptible-eval/evaluator/run
               session.clj:  229  nrepl.middleware.session/session-exec/session-loop
        SessionThread.java:   21  nrepl.SessionThread/run
```

The motivation here was to be an easy way to convert any `throw` to an `inc-and-throw!` and get error counting _without_ disturbing any thoughtful error handling surrounding it, and, in particular, to reduce diff in the common situation where the `throw` is in one branch of some conditional, where you would otherwise need to wrap it in `do` to insert the `inc!`. But as @metamben [rightly pointed out](https://github.com/metabase/metabase/pull/48598#discussion_r1799965415), most of the time you just want a single `try`/`catch` at some entry point to the module that's counting errors, so this wrapper is now even less useful than when I added it, and I'm happy to remove it if there are objections.

Reply:
I went ahead and removed `inc-and-throw!` for the following reasons

1. Opinions are divided (even I am not fully convinced tbh). When in doubt THROW IT OUT.
2. Recent updates to this PR mean there is now only a single call site, so value is questionable.
3. Easy to add back in the future if it proves useful other places, and API inertia means it's easier to add something to an API than it is to take it away later if we change our minds about it.